### ISSUE TITLE
Complex object selection

### ISSUE THREAD

**Issue by kryptus36** [go-pg/pg#575]
Given these types:

```
type User struct {
  tableName struct{} `sql:"users,alias:xuser"`
  Id int
  UserName string
}

type Domain struct {
  Id int
  ServerId int 
  Proxy bool
  UserId int
  User *User
}

type Server struct {
  Id int
  Fqdn string
  ProxyDomains []*Domain
}
```

Is it possible to fetch a Server with all associated ProxyDomains and each ProxyDomain having it's associated User populated?

**Comment by vmihailenco** [comment]
I suggest to try:

```
q.Model(&server).Column("ProxyDomains", "ProxyDomains.User").Select()
```

**Comment by zyukin** [comment]
I have the same issue. My model:

```go
type Comment struct {
        ID int
	UserID int 
	User User
	Comment string
	Data time.Time
}

type User struct {
	ID int
	Email string
	Password string
	Name string
	AvatarUID string `sql:",type:uuid"`
	Avatar Image
}

type Image struct {
	UID string `sql:",pk,type:uuid default uuid_generate_v4()"`
	Path string
}
```

When i try select
`db.Model(&list).Column("comment.*", "User", "User. Avatar").Select()`
I get panic
`panic serving 127.0.0.1:64184: reflect: call of reflect.Value.Field on zero Value`

Trace:
```
panic(0x148ade0, 0xc420519360)
	/usr/local/Cellar/go/1.8.3/libexec/src/runtime/panic.go:489 +0x2cf
reflect.Value.Field(0x0, 0x0, 0x0, 0x0, 0x151538f, 0x2, 0xffffffffffffffff)
	/usr/local/Cellar/go/1.8.3/libexec/src/reflect/value.go:759 +0x189
project/vendor/github.com/go-pg/pg/orm.fieldByIndex(0x0, 0x0, 0x0, 0xc420510b40, 0x1, 0x1, 0x151538f, 0x2, 0xc4205165b0)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/util.go:43 +0x77
project/vendor/github.com/go-pg/pg/orm.(*Field).ScanValue(0xc420512ab0, 0x0, 0x0, 0x0, 0xc4201ad000, 0x24, 0x200, 0xc4205191cb, 0x5)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/field.go:70 +0x63
project/vendor/github.com/go-pg/pg/orm.(*structTableModel).scanColumn(0xc4200cc4d0, 0x17, 0xc4205191d2, 0x3, 0xc4201ad000, 0x24, 0x200, 0xc4205191c0, 0x9, 0x11a3f01)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/model_table_struct.go:223 +0x13c
project/vendor/github.com/go-pg/pg/orm.(*structTableModel).scanColumn(0xc4200cc460, 0x17, 0xc4205191cb, 0xa, 0xc4201ad000, 0x24, 0x200, 0x0, 0x0, 0xc4201ecb20)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/model_table_struct.go:210 +0x2ff
project/vendor/github.com/go-pg/pg/orm.(*structTableModel).scanColumn(0xc4201322d0, 0x17, 0xc4205191c0, 0x15, 0xc4201ad000, 0x24, 0x200, 0xc4201ad000, 0x24, 0x200)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/model_table_struct.go:210 +0x2ff
project/vendor/github.com/go-pg/pg/orm.(*structTableModel).ScanColumn(0xc4201322d0, 0x17, 0xc4205191c0, 0x15, 0xc4201ad000, 0x24, 0x200, 0x0, 0x0)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/model_table_struct.go:199 +0x8e
project/vendor/github.com/go-pg/pg.readDataRow(0xc4201921b0, 0x1745220, 0xc4201322d0, 0xc4200ce280, 0x1a, 0x1a, 0x1a, 0x0)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/messages.go:584 +0x17c
project/vendor/github.com/go-pg/pg.readSimpleQueryData(0xc4201921b0, 0x1507080, 0xc4201322d0, 0x14dd060, 0xc420519000, 0xc420125e70)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/messages.go:634 +0x2b1
project/vendor/github.com/go-pg/pg.(*DB).simpleQueryData(0xc4201b0780, 0xc4201921b0, 0x1507080, 0xc4201322d0, 0x14dd060, 0xc420519000, 0xc420125e70, 0x1, 0x1, 0x10, ...)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/db.go:371 +0xe0
project/vendor/github.com/go-pg/pg.(*DB).Query(0xc4201b0780, 0x1507080, 0xc4201322d0, 0x14dd060, 0xc420519000, 0xc420125e70, 0x1, 0x1, 0x18, 0x11a8a01, ...)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/db.go:194 +0x163
project/vendor/github.com/go-pg/pg/orm.(*Query).query(0xc4201c26e0, 0x174ec00, 0xc4201322d0, 0x14dd060, 0xc420519000, 0xc4201322d0, 0x0, 0x0, 0xc4201ecf58)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/query.go:424 +0x285
project/vendor/github.com/go-pg/pg/orm.(*Query).Select(0xc4201c26e0, 0x0, 0x0, 0x0, 0xc4201c26e0, 0xc4205183c0)
	/Users/alex/Development/Go/src/project/vendor/github.com/go-pg/pg/orm/query.go:394 +0x10c
```

**Comment by kryptus36** [comment]
Just a guess, but:

type User struct {
	ID int
	Email string
	Password string
	Name string
	AvatarUID string `sql:",type:uuid"`
	Avatar *Image
}

(Change type of Avatar to pointer)

**Comment by zyukin** [comment]
I tried. Did not work.

**Comment by vmihailenco** [comment]
@zyukin should be fixed in v6.4.4
